Initializing PathAnnotator for ndim > 3 images fails

Currently, we have two test failures (see below), which indicate that initializing the 

I tracked it down and it looks like the the empty PathAnnotator gets initialized with a 3D points layer by default, irrespective of the dimensionality of the provided image layer.

The `__init__()` calls `PathAnnotator.set_layers()`
https://github.com/napari-threedee/napari-threedee/blob/2a063b9e151c40ea9285622755e4312ad519a1af/src/napari_threedee/annotators/paths/annotator.py#L71-L82

which calls `PathAnnotator._create_points_layer()`
https://github.com/napari-threedee/napari-threedee/blob/2a063b9e151c40ea9285622755e4312ad519a1af/src/napari_threedee/annotators/paths/annotator.py#L59-L62

which creates a 3D points layer
https://github.com/napari-threedee/napari-threedee/blob/2a063b9e151c40ea9285622755e4312ad519a1af/src/napari_threedee/data_models/paths.py#L80-L87
 
## Proposed fixes
1. We can just say that we only work with 3D data to start
2. We can add an ndim parameter
3. We can add an alternative constructor that is a class method on `N3DPaths` (e.g., `N3DPaths.create_empty_layer(ndim=4)`)

I am in favor of option 3 as it creates a clear special case. It could look something like this:

```python
class N3dPaths:
...
        @classmethod
        def create_empty_layer(cls, ndim: int=3):
                    # workaround for napari/napari#4213
                    dummy_data = np.zeros((ndim,))
                    n3d_paths = cls(data=[N3dPath(data=dummy_data])
                    layer = n3d_paths.as_layer()
                    layer.selected_data = {0}
                    layer.remove_selected()
                    return layer

```


**Report from tests that are failing**

```python
________________________________ test_add_point ________________________________
  
  spline_annotator = <napari_threedee.annotators.paths.annotator.PathAnnotator object at 0x7fd1b40b8040>
  
      def test_add_point(spline_annotator):
          points_layer = spline_annotator.points_layer
          label = napari_threedee.annotators.paths.constants.PATH_ID_FEATURES_KEY
      
          # start empty
          assert len(points_layer.data) == 0
      
          # add points, make sure filament id in feature table matches the annotator
  >       points_layer.add([1, 2, 3, 4])
  
  src/napari_threedee/annotators/_tests/test_spline_annotator.py:27: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  .tox/py39-linux/lib/python3.9/site-packages/napari/layers/points/points.py:1828: in add
      self.data = np.append(self.data, np.atleast_2d(coord), axis=0)
  <__array_function__ internals>:200: in append
      ???
  .tox/py39-linux/lib/python3.9/site-packages/numpy/lib/function_base.py:5499: in append
      return concatenate((arr, values), axis=axis)
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  args = ((array([], shape=(0, 3), dtype=float64), array([[1, 2, 3, 4]])),)
  kwargs = {'axis': 0}
  relevant_args = (array([], shape=(0, 3), dtype=float64), array([[1, 2, 3, 4]]))
  
  >   ???
  E   ValueError: all the input array dimensions except for the concatenation axis must match exactly, but along dimension 1, the array at index 0 has size 3 and the array at index 1 has size 4
  
  <__array_function__ internals>:200: ValueError
  _______________________________ test_get_colors ________________________________
  
  spline_annotator = <napari_threedee.annotators.paths.annotator.PathAnnotator object at 0x7fd17ae34b20>
  
      def test_get_colors(spline_annotator):
          """Test getting spline colors from the annotator."""
          points_layer = spline_annotator.points_layer
          spline_annotator.active_spline_id = 0
  >       points_layer.add([1, 2, 3, 4])
  
  src/napari_threedee/annotators/_tests/test_spline_annotator.py:41: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  .tox/py39-linux/lib/python3.9/site-packages/napari/layers/points/points.py:1828: in add
      self.data = np.append(self.data, np.atleast_2d(coord), axis=0)
  <__array_function__ internals>:200: in append
      ???
  .tox/py39-linux/lib/python3.9/site-packages/numpy/lib/function_base.py:5499: in append
      return concatenate((arr, values), axis=axis)
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
  
  args = ((array([], shape=(0, 3), dtype=float64), array([[1, 2, 3, 4]])),)
  kwargs = {'axis': 0}
  relevant_args = (array([], shape=(0, 3), dtype=float64), array([[1, 2, 3, 4]]))
  
  >   ???
  E   ValueError: all the input array dimensions except for the concatenation axis must match exactly, but along dimension 1, the array at index 0 has size 3 and the array at index 1 has size 4
  
  <__array_function__ internals>:200: ValueError
```

--- a/docs/examples/plugin/spline_annotator_plugin.py
+++ b/docs/examples/plugin/spline_annotator_plugin.py
@@ -45,7 +45,7 @@ viewer.add_image(
 
 # add plugin dock widget to viewer
 viewer.window.add_plugin_dock_widget(
-    plugin_name="napari-threedee", widget_name="sphere annotator"
+    plugin_name="napari-threedee", widget_name="path annotator"
 )
 
 # run napari

--- a/src/napari_threedee/annotators/_tests/test_spline_annotator.py
+++ b/src/napari_threedee/annotators/_tests/test_spline_annotator.py
@@ -26,22 +26,29 @@ def test_add_point(spline_annotator):
     # add points, make sure filament id in feature table matches the annotator
     points_layer.add([1, 2, 3, 4])
     assert len(points_layer.data) == 1
-    assert points_layer.features[label][0] == spline_annotator.active_spline_id
+    assert points_layer.features[label][0] == 0
 
-    # change filemanet_id in annotator, add point, check matches
-    spline_annotator.active_level_id = 534
+    # change filamanet_id in annotator, add point, check matches
     points_layer.add([2, 3, 4, 5])
-    assert points_layer.features[label][1] == spline_annotator.active_spline_id
+    assert points_layer.features[label][1] == 0
+
+    # create new path, make sure id advanced
+    spline_annotator.activate_new_path_mode()
+    points_layer.add([2, 3, 4, 5])
+    assert points_layer.features[label][2] == 1
 
 
 def test_get_colors(spline_annotator):
     """Test getting spline colors from the annotator."""
+    # add a point. the first spline gets id 0
     points_layer = spline_annotator.points_layer
-    spline_annotator.active_spline_id = 0
     points_layer.add([1, 2, 3, 4])
-    spline_annotator.active_spline_id = 1
+
+    # create a new spline, this advances the spline id to 1
+    spline_annotator.activate_new_path_mode()
     points_layer.add([2, 3, 4, 5])
 
     spline_colors = spline_annotator._get_path_colors()
     for spline_id in (0, 1):
+        # check that both spline ids are in the colors dict
         assert spline_id in spline_colors

--- a/src/napari_threedee/annotators/_tests/test_spline_annotator.py
+++ b/src/napari_threedee/annotators/_tests/test_spline_annotator.py
@@ -31,11 +31,13 @@ def test_add_point(spline_annotator):
     # change filamanet_id in annotator, add point, check matches
     points_layer.add([2, 3, 4, 5])
     assert points_layer.features[label][1] == 0
+    assert spline_annotator.active_path_id == 0
 
     # create new path, make sure id advanced
     spline_annotator.activate_new_path_mode()
     points_layer.add([2, 3, 4, 5])
     assert points_layer.features[label][2] == 1
+    assert spline_annotator.active_path_id == 1
 
 
 def test_get_colors(spline_annotator):

--- a/src/napari_threedee/annotators/paths/annotator.py
+++ b/src/napari_threedee/annotators/paths/annotator.py
@@ -58,8 +58,8 @@ class PathAnnotator(N3dComponent):
 
     def _create_points_layer(self) -> Points:
         from napari_threedee.data_models import N3dPaths
-        layer = N3dPaths(data=[]).as_layer()
-        return layer
+        ndim = self.image_layer.data.ndim
+        return N3dPaths.create_empty_layer(ndim=ndim)
 
     def _create_shapes_layer(self) -> Shapes:
         return Shapes(

--- a/src/napari_threedee/annotators/paths/annotator.py
+++ b/src/napari_threedee/annotators/paths/annotator.py
@@ -35,6 +35,10 @@ class PathAnnotator(N3dComponent):
 
         self.enabled = enabled
 
+    @property
+    def active_path_id(self):
+        return self.points_layer.current_properties[PATH_ID_FEATURES_KEY][0]
+
     def activate_new_path_mode(self, event=None) -> None:
         if self.points_layer is None:
             return

--- a/src/napari_threedee/data_models/paths.py
+++ b/src/napari_threedee/data_models/paths.py
@@ -128,6 +128,31 @@ class N3dPaths(N3dDataModel):
         n3d_zarr.attrs[ANNOTATION_TYPE_KEY] = PATH_ANNOTATION_TYPE_KEY
         n3d_zarr.attrs[PATH_ID_FEATURES_KEY] = list(self.spline_ids)
 
+    @classmethod
+    def create_empty_layer(cls, ndim: int):
+        """Create an empty path annotator points layer.
+
+        Parameters
+        ----------
+        ndim : int
+            The dimensionality of the empty points layer.
+            Generally, this should match the image layer being
+            annotated.
+
+        Returns
+        -------
+        layer : napari.layers.Points
+            The napari Points layer initialized for path annotation.
+        """
+        # workaround for napari/napari#4213
+        dummy_data = np.zeros((ndim,))
+        n3d_paths = cls(data=[N3dPath(data=dummy_data)])
+        layer = n3d_paths.as_layer()
+        layer.selected_data = {0}
+        layer.remove_selected()
+        return layer
+
+
     def __getitem__(self, idx: int) -> N3dPath:
         return self.data[idx]
 

